rcl/src/rcl/logging_rosout.c

```diff
@@ -438,20 +438,20 @@ rcl_logging_rosout_add_sublogger(
   rosout_map_entry_t entry;
   rosout_sublogger_entry_t sublogger_entry;
 
+  RCL_CHECK_ARGUMENT_FOR_NULL(logger_name, RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(sublogger_name, RCL_RET_INVALID_ARGUMENT);
+  rcutils_ret_t rcutils_ret = rcutils_hash_map_get(&__logger_map, &logger_name, &entry);
+  if (RCUTILS_RET_NOT_FOUND == rcutils_ret) {
+    return RCL_RET_OK;
+  }
```

Comment:
I am not sure if this is expected behavior... IMO this still needs to fail? cz it failed to add sub-logger to the hash map? this situation could happen with certain condition such as disabling rosout or node might not exist. no matter it is, it failed to add the sub-logger, so user should expect the failure for that?
how about dealing with case in client libraries, we can check the error code, and if that is RCUTILS_RET_NOT_FOUND with warning?

Reply:
Thanks.
What I did here is similar to https://github.com/ros2/rcl/blob/11103d750af1dc537d229a7e99bb05180c890088/rcl/src/rcl/logging_rosout.c#L431-L433 that if there is uninitialized.

I think your suggestion makes sense. I'll update it.